chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/state/JoinedRoomsStateDiffer.kt

```diff
@@ -44,4 +47,16 @@ internal class JoinedRoomsStateDiffer(private val stateGetter: GetState<State>)
     ): List<LeftRoom> =
         (currentRooms.keys.toSet() - newRooms.map { it.id }.toSet())
             .map { LeftRoom(roomId = it) }
+
+    private fun unreadCountActions(
```

Comment:
differ should not be concerned with unread counts, the event should be always dispatched by the dispatcher for each read state it receives (both for specific ReadStateUpdated event, and new initial state, by "new" I mean when the state exists already -- reconect scenario). reducer should check if unread count has changed and apply the change only then

Reply:
yep I think that makes the best sense. for `ReconnectJoinedRoom` let's keep optional 'unreadCount` so that we get all the data atomically to run notification to the client code. just `UnreadCountReceived` events should be after the room actions dispatched so that no unread count is added before a joined room is added